ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -1629,7 +1630,7 @@ synchronized boolean isEmpty() {
     }
 
     /**
-     *  get the keys corresponding to the missing messages in local store
+     *  Get the keys corresponding to the missing messages in local store
      */
     Set<StoreKey> getMissingStoreKeys() {
```

Comment:
Should this method be synchronized too? It seems like missingStoreMessages can be mutated and using a stream is just like iterating through it and subject to a ConcurrentModificationException.

Reply:
+ 1, good point.